sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -116,6 +117,13 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
       val rdd = v1Relation.buildScan()
       val unsafeRowRDD = DataSourceStrategy.toCatalystRDD(v1Relation, output, rdd)
 
+      val tableIdentifier = v2Relation.identifier match {
```

Comment:
we can import `CatalogV2Implicits._` and then do `v2Relation.identifier.asMultipartIdentifier`

Reply:
Nice, I did not know for that, we can then avoid creation of new Object for unapply